### DISCUSSION TITLE
Conditional data.get() from specific trials

### DISCUSSION THREAD

**Discussion by iwe90** [jspsych/jsPsych#1834] [Q&A]
Hi everyone,

I'm quite new to jspsych and I have run into a problem while coding a simple online questionnaire.

The entire structure of the questionnaire consists of if nodes, because the appearance of all following questions depends on the responses in the first survey-likert type item and/or following items.

My problem now is that I can't simply use 
`var data = jsPsych.data.get().last(1).values()[0];`
to access the responses from the previous trials because as I understand, last() needs the exact number of previous trials (it breaks down if I put more than have been collected)- which is not possible because some people get more questions than others, depending on their responses.

Using
`var first_qu = jsPsych.data.get().filter({trial: 'survey-likert'})//.values()[0];`
also doesn't work for some reason and isn't helpful for the entire experiment,

neither does 
`var data = jsPsych.data.getDataByTimelineNode(0.0-1.0)`

Can anyone suggest a different way to access the individual question responses without referring to their sequence in the data frame?

Thanks & best
Ivonne

**Comment by jodeleeuw** [comment]
Hi Ivonne,

You can use the [data parameter](https://www.jspsych.org/overview/data/#adding-data-to-a-particular-trial-or-set-of-trials) of the trial to add a unique identifier for each question. Then you can use the [.filter()](https://www.jspsych.org/core_library/jspsych-data/#filter) method to get data from the trial that matches that identifier.

```js
var trial = {
	type: 'survey-text',
	questions: [{prompt: 'What is your name?'}],
	data: {
		question_id: 'name'
	}
}

jsPsych.data.get().filter({question_id: 'name'})
```

**Reply by iwe90** [reply]
Cool, thanks for the quick reply!
I'm currently trying this and I still need the ".values()[0]" at the end, though, right?

`        var if_node_spr1 = {
            timeline: [quest_pic_spr_1],
            conditional_function: function() {
            // get the data from the previous trial,
            // and check which key was pressed
                var data = jsPsych.data.get().filter({question_id: 'quest_pic_1'}).values()[0]
                if(data.response.Sprachlich == '0'){
                    return false;
                } else {
                    return true;
                }
            }
        }
        timeline.push(if_node_spr1);`

**Reply by jodeleeuw** [reply]
Yes. The filter function will return the [data collection](https://www.jspsych.org/core_library/jspsych-data/#datacollection) and then you would need `.values()[0]` to get the raw data for a trial.
